### ISSUE TITLE
Error in 'run script to train and test model'

### ISSUE THREAD

**Issue by mysope** [smallcorgi/Faster-RCNN_TF#79]
"6.Run script to train and test model"
After the training is finished, the test does not proceed and the following error occurs.

------
+ python ./tools/test_net.py --device gpu --device_id 0 --weights --imdb voc_2007_test --cfg experiments/cfgs/faster_rcnn_end2end.yml --network VGGnet_test
------

It says that the VGGnet_fast_rcnn_iter_70000.ckpt file is recorded in the snapshot unlike the log below, but I can not find the file. 

-------
Wrote snapshot to: /home/nwkim/tensorflow-sample/Faster-RCNN_TF/output/faster_rcnn_end2end/voc_2007_trainval/VGGnet_fast_rcnn_iter_70000.ckpt
-------

I have just seen other files, such as the file directory below, and I've renamed it to test, but it will not run. (It will be executed by inserting the downloaded VGGnet_fast_rcnn_iter_70000.ckpt file that was used when executing the demo)

----
<File Directoy>

nwkim@hypo:~/tensorflow-sample/Faster-RCNN_TF/output/faster_rcnn_end2end/voc_2007_trainval$ ls
.....
VGGnet_fast_rcnn_iter_70000.ckpt.data-00000-of-00001
VGGnet_fast_rcnn_iter_70000.ckpt.index
VGGnet_fast_rcnn_iter_70000.ckpt.meta
----

What is the problem??

----------------------------------------------------------------------

iter: 70000 / 70000, total loss: 0.6129, rpn_loss_cls: 0.0193, rpn_loss_box: 0.2729, loss_cls: 0.1254, loss_box: 0.1954, lr: 0.000100
speed: 0.642s / iter
Wrote snapshot to: /home/nwkim/tensorflow-sample/Faster-RCNN_TF/output/faster_rcnn_end2end/voc_2007_trainval/VGGnet_fast_rcnn_iter_70000.ckpt
done solving

real	762m15.862s
user	516m9.624s
sys	67m31.156s
+ set +x
grep: experiments/logs/faster_rcnn_end2end_VGG16_.txt.2017-02-07_08-59-31: No such file or directory
+ python ./tools/test_net.py --device gpu --device_id 0 --weights --imdb voc_2007_test --cfg experiments/cfgs/faster_rcnn_end2end.yml --network VGGnet_test
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:128] successfully opened CUDA library libcurand.so locally
voc_2007_train
voc_2007_val
voc_2007_trainval
voc_2007_test
kitti_train
kitti_val
kitti_trainval
kitti_test
nthu_71
nthu_370
usage: test_net.py [-h] [--device DEVICE] [--device_id DEVICE_ID]
                   [--def PROTOTXT] [--weights MODEL] [--cfg CFG_FILE]
                   [--wait WAIT] [--imdb IMDB_NAME] [--comp]
                   [--network NETWORK_NAME]
test_net.py: error: argument --weights: expected one argument

**Comment by ArturoDeza** [comment]
I believe I figures this out.

Tensorflow has been updated and is saving 3 types of check points (instead of only 1). You will need to change the line in the `FASTER-RCNN_TF/fast_rcnn/train.py` :
`self.saver = tf.train.Saver(max_to_keep=100)`
to:
`self.saver = tf.train.Saver(max_to_keep=100,write_version=saver_pb2.SaverDef.V1)`

And add this in the same train.py file preamble:
`from tensorflow.core.protobuf import saver_pb2`

See this post for more details. Let me know if this works.
http://stackoverflow.com/questions/41048819/how-to-restore-a-model-by-filename-in-tensorflow-r12

**Comment by mysope** [comment]
@ArturoDeza Cool! It works well. Thanks a lot. :+1:

**Comment by Bigwode** [comment]
WTF ? It means that I should train again ???  but when I run the demo,  the three saved ckpt. file can work  well.

**Comment by Bigwode** [comment]
WARNING:tensorflow:TensorFlow's V1 checkpoint format has been deprecated.
WARNING:tensorflow:Consider switching to the more efficient V2 format:
WARNING:tensorflow:   `tf.train.Saver(write_version=tf.train.SaverDef.V2)`
WARNING:tensorflow:now on by default.
失效了，不能用了，尴尬

**Comment by YCIrving** [comment]
@Bigwode Just comment the loop in test_net.py
`while not os.path.exists(args.model) and args.wait:`
`print('Waiting for {} to exist...'.format(args.model))`
`time.sleep(10)`
